.github/workflows/build-test.yml

```diff
@@ -61,3 +61,85 @@ jobs:
     - name: Rust linter
       # make this command fail if cargo fmt had to make changes
       run: cargo fmt && git diff-index --exit-code HEAD
+
+  # Linting job permissive (cargo-clippy) - completes and puts warnings inline in PR
+
+  lint:
+    runs-on: ubuntu-latest
+
+    needs: [build]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    # Linting job permissive > Cache steps
+
+    - name: Cache cargo registry
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo/registry
+        key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
+
+    - name: Cache cargo index
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo/git
+        key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
+
+    - name: Cache cargo build
+      uses: actions/cache@v1
+      with:
+        path: target
+        key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.toml') }}
+
+    # Linting job permissive > Install and run clippy steps
+
+    - name: Install clippy
+      run: rustup component add clippy
+
+    # Run clippy first time to get warnings inserted inline in PR
+    - uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: --all-targets --all-features
+
+  # Linting job strict (cargo-clippy)
+
+  lint_strict:
+    runs-on: ubuntu-latest
+
+    needs: [build]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    # Linting job strict > Cache steps
+
+    - name: Cache cargo registry
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo/registry
+        key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
+
+    - name: Cache cargo index
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo/git
+        key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
+
+    - name: Cache cargo build
```

Comment:
If you don't cache the cargo build, does that avoid the need for the cargo clean?

Reply:
Good catch -- without the caching step (which is restoring the cache at the beginning), cargo clippy doesn't need the cargo clean step in order to [still fail on warnings](https://github.com/echeran/icu4x/runs/827887615?check_suite_focus=true).  I updated accordingly.  This holds true, at least on CI, because each Github Actions job is run on a fresh VM image.  (Hence, the need to start off each job with `actions/checkout@v2` to clone the repo in the job's VM.)